src/runtime_src/core/tools/xbmgmt2/ReportPlatform.cpp

```diff
@@ -222,6 +222,9 @@ ReportPlatform::writeReport( const xrt_core::device * device,
                              std::iostream & output) const
 {
   boost::property_tree::ptree pt;
+  bool is_mfg = xrt_core::device_query<xrt_core::query::is_mfg>(device);
+  Flasher f(0);

```

Comment:
This is not good... We should never have magic numbers in the code and also, this will always open device 0. This doesn't work for multi-device systems. You need to make changes in 2 places:
1. src/runtime_src/core/tools/common/XBUtilities.cpp L385
 ```
//if factory mode
    if (is_mfg) {
      std::string vbnv = "xilinx_" + xrt_core::device_query<xrt_core::query::board_name>(device) + "_GOLDEN";
                          <...... ADD THE VERSION HERE.....>
      pt_dev.put("vbnv", vbnv);
    }
```
2. src/runtime_src/core/tools/xbmgmt2/ReportPlatform.cpp L133

```
if(xrt_core::device_query<xrt_core::query::is_mfg>(device)) { // golden
    std::string board_name = xrt_core::device_query<xrt_core::query::board_name>(device);
    std::string vbnv = "xilinx_" + board_name + "_GOLDEN";
                          <...... ADD THE VERSION HERE.....>
    pt_current_shell.put("vbnv", vbnv);
  }
```

`mfg_ver` device_query is available for use, so please use that instead of genOnBoardDSA()

Reply:
We need to make sure that all commands that show us the platform name (vnbv) show the golden version as well. The CR asks to fix `xbmgmt examine -r platform`, but we also have `xbmgmt examine -r host` which will show the platform name. Please test both these commands :)